File: Programmers/hyeonyoung/2개 이하로 다른 비트.cpp
#include <string>
#include <vector>

using namespace std;

vector<long long> solution(vector<long long> numbers)
{
    vector<long long> answer;
    for (long long x : numbers)
    {
        long long ans = x;
        for (int i = 0; i < 64; ++i)
        {
            if ((ans & ((long long)1 << i)) == 0)
            {
                ans |= ((long long)1 << i);
                if (i > 0)
                {
                    ans -= ((long long)1 << (i - 1));
                }
                break;
            }
        }

        answer.push_back(ans);
    }
    return answer;
}

File: Programmers/suah/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    for n in numbers: 
        b = list('0' + bin(n)[2:])
        idx = ''.join(b).rfind('0')
        b[idx] = '1'
        
        # 홀수인 경우
        if n % 2 == 1: 
            b[idx+1] = '0'
            
        answer.append(int(''.join(b), 2))
    return answer
